### PR TITLE
Jenkins jobs for Ant build

### PR DESCRIPTION
**Pull request by rfelcman**
This is patch related with migration to new cloud based Eclipse.org build infrastructure and
" The end of an era: shell access".
There are Jenkins pipelines 
This jobs contains two pipelines:
1. Build part - _build.groovy, init.sh, build.sh, publish_nightly.sh ..._
   Build project, execute test, publish to nightly builds.
   These files are common for nightly and ci builds. There is parameter CONTINUOUS_BUILD specified (true - continuous build, false - full nightly build)  
2. Promotion part - _promote.groovy, promote_init.sh, promote.sh ..._
   Publish Maven artifacts to the Maven repository (OSS staging). This is Jenkins replacement of _promote.sh_ called in shell available at build.eclipse.org infrastructure.
  Due this part, there are modifications in some current files in the project root. They are:
_autobuild.xml_ : Switch to Maven 3 and some check. _promote.sh_ used Maven 2 and JDK 6 before.
_pom.xml.template_ : EE4J parent pom added . This pom file is used as a template for poms of Maven artifacts published to the Maven Central.
_ uploadToNexus.xml _ : There is change how Maven is called. Before this change tool _maven-ant-tasks-2.1.3.jar_ was used, but there was troubles with GPG call (required interactive env). After this fix command line call is used.

Signed-off-by: Radek Felcman <radek.felcman@oracle.com>

### CODE REVIEW

**Comment by lukasj** [comment]
@dazey3 FYI

File: etc/jenkins/setEnvironment.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# Contributors:
+#  - rfelcman - 16 August 2019 - Initial implementation
+#****************************************************************************************
+
+# This script sets environment variables used by other scripts in eclipselink.releng.
+
+ANT_HOME_DEFAULT=/opt/java/ant
+M2_HOME_DEFAULT=/opt/java/maven
+HOME_DIR_DEFAULT=${WORKSPACE}
+JAVA_HOME_DEFAULT=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.222.b10-0.fc30.x86_64
```
Comment (dazey3) [inline etc/jenkins/setEnvironment.sh]: Should we extract this out to a property file?

**Comment by dazey3** [comment]
@lukasj Thanks for keeping me involved :) So, who runs these scripts? Are these just for the builds running out on https://hudson.eclipse.org/eclipselink/ ?

**Comment by Tomas-Kraus** [comment]
They shall be used on jenkins.eclipse.org - their new build infrastructure. It will replace current ci.eclipse.org/eclipselink which is going to be retired.

**Comment by lukasj** [comment]
@dazey3 for background: https://www.eclipse.org/lists/cross-project-issues-dev/msg16922.html and https://bugs.eclipse.org/bugs/show_bug.cgi?id=546431

File: etc/jenkins/antbuild.xml [hunk 2]
```diff
@@ -0,0 +1,376 @@
+<!--
+
+    Copyright (c) 1998, 2019 Oracle and/or its affiliates. All rights reserved.
```
Comment (lukasj) [inline etc/jenkins/antbuild.xml]: new file, so 2019 only

**Comment by rfelcman** [comment]
Let me tune Jenkins junit plugin settings.

**Pull request merged**